styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -127,14 +128,27 @@ public QueuedStateManager(
 
   @Override
   public CompletableFuture<Void> receive(Event event) throws IsClosedException {
+    ensureRunning();
+    // Read state counter at enqueueing time
+    final Optional<RunState> currentRunState = getActiveState(event.workflowInstance());
```

Comment:
This makes all callers of `receive` block on reading the active state from datastore. I worry that this causes problems like e.g. the k8s pod watcher not being able to keep up with the rate of events as it is single threaded.

Reply:
This is partially addressed by https://github.com/spotify/styx/pull/431 where we send in counter explicitly.